### ISSUE TITLE
How shall we revert master?

### ISSUE THREAD

**Issue by mih** [datalad/datalad#2022]
Just reset to `0.9`? Revert commits? Any preference?

I would reset it to 0.9 in a few days, if nobody has a better idea.

**Comment by yarikoptic** [comment]
I think we simply shouldn't -- that would bring more confusion etc and wouldn't solve any problem.
What we should is to make master usable by mortal human beings and more or less consistent with how it was working before, e.g. may be by default do index/search only across datasets metadata.  Should make it reasonably fast to index/search, while making it up to the user to create a full index with per-file based meta-data for more advanced querying.  I was also thinking about leaving 'search' a dummy one (as before) and either make `query` or `search --query` to do all the fancy globbing/selection of level, etc.  not sure if that is achievable though since didn't play with search enough yet

**Comment by mih** [comment]
Sure,  let's do that in a feature branch dedicated to developing that feature. Meanwhile we reset master to let people use it for development of any kind. In particular the kind of development that is very much needed in master, but is happening in 0.9. Such a switch takes a few seconds. Developing a feature to mask a feature to move towards a better feature in the future just delays things and is not a timely fix to the current mess.

**Comment by yarikoptic** [comment]
> Sure, let's do that in a feature branch dedicated to developing that feature

that is how all those fixes appeared -- in feature branches.

> Meanwhile we reset master to let people use it for development of any kind.

once again -- branches are just names. We would need a hard reset which is "evil" and will cause confusion etc.  People are welcome to develop on any branch they feel requiring development.  If we agree that 0.9.x should be such a branch -- just make it a default one, no reset needed

> kind of development that is very much needed in master, but is happening in 0.9

0.9 has no development as far as I see -- only fixes.  And it is a matter of an easy merge into master to absorb them.  If you want, we could position all fixes against master but then I would need manually to cherry pick etc into 0.9.x -- imho much more inferior approach in comparison to "just one more merge"

If you keep referring to the feature of tests dedicated "$HOME"  -- took me days of round trips with travis to iron it out despite it being simple but me doing it in parallel to other things, and it takes seconds to do a merge into master.  Sorry that I made it so hard for you.

**Comment by mih** [comment]
I don't think that this is an accurate description of the situation. The test thingie was a long requested feature. It was developed against 0.9, not master. The PR against master was initiated by someone other than the developer making the feature. When the tests against master actually completes without error (not yet done), we will know whether it was an easy merge. But we already know that this is "hope", "should be", or something along those lines, because travis has never thrown the diff against master during the actual development phase.

**Comment by yarikoptic** [comment]
uff.. ok -- for such features,I could send two PRs if it would make you happy(ier).
Side -- actually I see the benefit of having those fixes going against 0.9.x as the one most used, thus verifying they do not cause obvious regression, and then a merge of 0.9.x into master -- as to introduce fixes and see if they do not break anything what was changed in master. 
I will conclude my opinion on the issue with this post -- if you feel that doing some reset would improve things -- go ahead, but it goes against my opinion on the matter and would make my life a  bit harder (in comparison to a single additional PR/merge as it is now with benefit outlined above).  What we should aim is to make master releasable, and no renaming/resetting would make that happen on its own.

**Comment by mih** [comment]
I failed to make my point it seems. I continue to be under the impression that the dev model has changed substantially. BTW: each commit in the prev. merge-back into master was a fix that was needed in master too.

**Comment by mih** [comment]
One last FTR: I think it is not a good idea to use a stable release as the primary development target -- I also cannot think of any project that does that. All it can lead to is stagnation. The majority of the development force will run the stable release, and the actual development gets little attention. There is a reason why Debian and the like recommend its developers to actually use the development version.

**Comment by yarikoptic** [comment]
I promise to do new development on top of master (how I did e.g. #1942).

**Comment by mih** [comment]
Pinky promise?

**Comment by yarikoptic** [comment]
yes, always!
... but (applicable) fixes to 0.9.x ;)
